bin/check-rabbitmq-consumer-utilisation.rb

```diff
@@ -0,0 +1,167 @@
+#!/usr/bin/env ruby
+
+# Check RabbitMQ consumer utilisation
+# ===
+#
+# DESCRIPTION:
+# This plugin checks the consumer utilisation percentage.
+# The fraction of time in which the queue is able to immediately deliver
+# messages to consumer. If this number drops in percentage this may result
+# in slower message delivery and indicate issues with the queue.
+#
+# Seeded from check-rabbitmq-consumers.rb
+#
+# PLATFORMS:
+#   Linux, BSD, Solaris
+#
+# DEPENDENCIES:
+#   RabbitMQ rabbitmq_management plugin
+#   gem: sensu-plugin
+#   gem: carrot-top
+#
+# LICENSE:
+# Copyright 2014 Daniel Kerwin <d.kerwin@gini.net>
+# Copyright 2014 Tim Smith <tim@cozy.co>
+# Copyright 2018 Mike Murray <37150283+monkey670@users.noreply.github.com>
+#
+# Released under the same terms as Sensu (the MIT license); see LICENSE
+# for details.
+
+require 'sensu-plugin/check/cli'
+require 'carrot-top'
+require 'inifile'
+
+# main plugin class
+class CheckRabbitMQConsumerUtilisation < Sensu::Plugin::Check::CLI
+  option :host,
+         description: 'RabbitMQ management API host',
+         long: '--host HOST',
+         default: 'localhost'
+
+  option :port,
+         description: 'RabbitMQ management API port',
+         long: '--port PORT',
+         proc: proc(&:to_i),
+         default: 15_672
+
+  option :ssl,
+         description: 'Enable SSL for connection to the API',
+         long: '--ssl',
+         boolean: true,
+         default: false
+
+  option :username,
+         description: 'RabbitMQ management API user',
+         long: '--username USER',
+         default: 'guest'
+
+  option :password,
+         description: 'RabbitMQ management API password',
+         long: '--password PASSWORD',
+         default: 'guest'
+
+  option :queue,
+         description: 'Comma separated list of RabbitMQ queues to monitor.',
+         long: '--queue queue_name',
+         proc: proc { |q| q.split(',') }
+
+  option :exclude,
+         description: 'Comma separated list of RabbitMQ queues to NOT monitor.  All others will be monitored.',
+         long: '--exclude queue_name',
+         proc: proc { |q| q.split(',') }
+
+  option :regex,
+         description: 'Treat the --queue flag as a regular expression.',
+         long: '--regex',
+         boolean: true,
+         default: false
+
+  option :warn,
+         short: '-w CONSUMER_UTILISATION',
+         long: '--warn CONSUMER_UTILISATION',
+         proc: proc(&:to_f),
+         description: 'WARNING consumer utilisation threshold',
+         default: 0.9
+
+  option :critical,
+         short: '-c CONSUMER_UTILISATION',
+         long: '--critical CONSUMER_UTILISATION',
+         description: 'CRITICAL consumer utilisation threshold',
+         proc: proc(&:to_f),
+         default: 0.5
+
+  option :ini,
+         description: 'Configuration ini file',
+         short: '-i',
+         long: '--ini VALUE'
+
+  def rabbit
```

Comment:
I think you can re-use this logic [here](https://github.com/sensu-plugins/sensu-plugins-rabbitmq/blob/5.0.0/lib/sensu-plugins-rabbitmq/rabbitmq.rb#L52)